src/Element.js

```diff
@@ -0,0 +1,341 @@
+			children[index] = from(arguments[i], index)
+		}
+		if (uuid < Constant.node) {
+			return iterable(Constant.fragment, [empty(Constant.key), element, empty(Constant.key)], key)
+		}
+	}
+
+	return element
+}
+
+/**
+ * @param {*} value
+ * @param {number} index
+ * @return {object}
+ */
+export function from (value, index) {
+	if (value != null) {
+		if (value.constructor === Constant.element) {
+			return value
+		} else {
+			switch (typeof value) {
+				case 'number':
+				case 'string':
+					return text(value, index)
+				case 'function':
+				case 'object':
+					if (Utility.iterable(value)) {
+						if (index === Constant.key) {
+							return iterable(Constant.fragment, value.map(from), Utility.hash(index))
+						} else {
+							return iterable(Constant.iterable, value.map(from), Utility.hash(index))
+						}
+					} else {
+						return create(value, Constant.object)
+					}
+			}
+		}
+	}
+
+	return empty(index)
+}
+
+/**
+ * @param {*} value
+ * @return {number}
+ */
+export function identity (value) {
+	switch (typeof value) {
+		case "string":
+			return Constant.node
+		case 'function':
+			return Constant.component
+		case 'number':
+			return Constant.fragment
+		case 'object':
+			return Utility.thenable(value) ? Constant.thenable : Constant.portal
+	}
+}
+
+/**
+ * @param {*} value
+ * @return {boolean}
+ */
+export function properties (value) {
+	if (value) {
+		if (typeof value === 'object') {
+			switch (value.constructor) {
+				case Object:
+					return true
+				case Constant.element:
+					break
+				default:
+					return !Utility.iterable(value)
+			}
+		}
+	}
+
+	return false
+}
+
+/**
+ * @param {*} value
+ * @return {string}
+ */
+export function display (value) {
+	switch (typeof value) {
+		case 'function':
+			return display(value[Constant.displayName] || value.name)
+		case 'object':
+			if (valid(value)) {
+				return display(value.type)
+			}
+		default:
+			return value.toString()
+	}
+}
+
+/**
+ * @param {object} element
+ * @param {object?} value
+ * @return {object}
+ */
+export function defaults (element, value) {
+	if (value) {
+		if (typeof value === 'function') {
+			defaults(element, value.call(element.type, element.props))
+		} else {
+			for (var key in value) {
+				if (element.props[key] === undefined) {
+					element.props[key] = value[key]
+				}
+			}
+		}
+	}
+
+	return element
+}
+
+/**
+ * @param {*} element
+ * @return {boolean}
+ */
+export function valid (element) {
+	return element != null && element.constructor === Constant.element
+}
+
+/**
+ * @param {object} value
+ * @param {...*?}
+ * @return {object}
+ */
+export function clone (element) {
+	return defaults(create.apply(null, [element.type].concat([].slice.call(arguments, 1))), element.props)
+}
+
+/**
+ * @param {object} element
+ * @param {number} from
+ * @return {object}
+ */
+export function parent (element, from) {
+	if (element.uuid < Constant.portal) {
+		if (element.uuid !== -from) {
```

Comment:
The constants(enums) for element `uuid` and `from` are aligned such that this is true only in one scenario: When we are coming from `create` and this is a fragment element. That is given the stateful nature of fragments in  native land we shadow the parent when not specifically in the initial mount phase since that is where it is implied they are if branch is reached. This might need to change depending on the new special element `offscreen` that doesn't necessarily want to be exposed when rendering offscreen.

Reply:
I don't follow
looks like its called from `remove`, `append`, `insert`, not `create`

also, a specific check for `fragment` type would help the code read (that is, if I understood the first sentence correctly)

Reply:
The idea is that `append`, `insert` and remove, would just pass the value on from the respective sources, the sources of which come from `Commit.mount` and `Commit.create`.

But yes it seems a better check is needed here.

Another implementation of this is if we could instead of eagerly setting `element.active` in `Element.constructor` we could do it in `Commit.create` after processing children. That way we can ascertain with proof wether the parent is coming from an update phase or a create phase(where parent.active would be Constant.idle or anything `< Constant.active`.